tests/racecontrol_test.py

```diff
@@ -16,79 +16,93 @@
+    ]
+
+    assert expected == racecontrol.available_pipelines()
 
-        racecontrol.run(cfg)
 
-        mock_pipeline.assert_called_once_with(cfg)
+def test_prevents_running_an_unknown_pipeline():
+    cfg = config.Config()
+    cfg.add(config.Scope.benchmark, "race", "pipeline", "invalid")
+    cfg.add(config.Scope.benchmark, "mechanic", "distribution.version", "5.0.0")
 
-        del racecontrol.pipelines[test_pipeline_name]
+    with pytest.raises(exceptions.SystemSetupError) as ex:
+        racecontrol.run(cfg)
+
+    assert re.match(r"Unknown pipeline \[invalid]. List the available pipelines with [\S]+? list pipelines.",
+                    ex.value.message)
 
-    @mock.patch.dict(os.environ, {"RALLY_RUNNING_IN_DOCKER": "true"})
-    def test_fails_without_benchmark_only_pipeline_in_docker(self):
-        mock_pipeline = mock.Mock()
-        test_pipeline_name = "unit-test-pipeline"
-        racecontrol.Pipeline("unit-test-pipeline", "Pipeline intended for unit-testing", mock_pipeline)
-        cfg = config.Config()
-        cfg.add(config.Scope.benchmark, "race", "pipeline", "unit-test-pipeline")
 
-        with self.assertRaises(exceptions.SystemSetupError) as ctx:
-            racecontrol.run(cfg)
+def test_passes_benchmark_only_pipeline_in_docker(running_in_docker, benchmark_only_pipeline):
+    cfg = config.Config()
+    cfg.add(config.Scope.benchmark, "race", "pipeline", "benchmark-only")
 
-        self.assertEqual(
-            "Only the [benchmark-only] pipeline is supported by the Rally Docker image.\n"
-            "Add --pipeline=benchmark-only in your Rally arguments and try again.\n"
-            "For more details read the docs for the benchmark-only pipeline in "
-            "https://esrally.readthedocs.io/en/latest/pipelines.html#benchmark-only\n",
-            ctx.exception.args[0])
-        del racecontrol.pipelines[test_pipeline_name]
+    racecontrol.run(cfg)
 
-    def test_runs_a_known_pipeline(self):
-        mock_pipeline = mock.Mock()
-        test_pipeline_name = "unit-test-pipeline"
+    benchmark_only_pipeline.target.assert_called_once_with(cfg)
 
-        racecontrol.Pipeline("unit-test-pipeline", "Pipeline intended for unit-testing", mock_pipeline)
 
-        cfg = config.Config()
-        cfg.add(config.Scope.benchmark, "race", "pipeline", "unit-test-pipeline")
-        cfg.add(config.Scope.benchmark, "mechanic", "distribution.version", "")
+def test_fails_without_benchmark_only_pipeline_in_docker(running_in_docker, unittest_pipeline):
+    cfg = config.Config()
+    cfg.add(config.Scope.benchmark, "race", "pipeline", "unit-test-pipeline")
 
+    with pytest.raises(exceptions.SystemSetupError) as ex:
         racecontrol.run(cfg)
 
-        mock_pipeline.assert_called_once_with(cfg)
+    assert ex.value.message == "Only the [benchmark-only] pipeline is supported by the Rally Docker image.\n" \
```

Comment:
Same suggestion as above, for leveraging [match](https://docs.pytest.org/en/stable/reference.html?highlight=raises#pytest.raises), and using `re.escape()` since this is a literal match i.e.

```
    with pytest.raises(
            exceptions.SystemSetupError,
            match=re.escape(
                "Only the [benchmark-only] pipeline is supported by the Rally Docker image.\n"
                "Add --pipeline=benchmark-only in your Rally arguments and try again.\n"
                "For more details read the docs for the benchmark-only pipeline in "
                "https://esrally.readthedocs.io/en/latest/pipelines.html#benchmark-only\n"
            )):
        racecontrol.run(cfg)
```

Reply:
Nice, thanks for the improvement! I've pushed this in https://github.com/elastic/rally/pull/1172/commits/dff0ee0fa9b109627dfbca1beec71435337e0d19.